src/main/drivers/dshot.c

```diff
@@ -145,17 +145,22 @@ FAST_DATA_ZERO_INIT dshotTelemetryState_t dshotTelemetryState;
 FAST_DATA_ZERO_INIT static pt1Filter_t motorFreqLpf[MAX_SUPPORTED_MOTORS];
 FAST_DATA_ZERO_INIT static float motorFrequencyHz[MAX_SUPPORTED_MOTORS];
 FAST_DATA_ZERO_INIT static float minMotorFrequencyHz;
+#if defined(USE_DSHOT_TELEMETRY) || defined(USE_ESC_SENSOR)
 FAST_DATA_ZERO_INIT static float erpmToHz;
+#endif
 FAST_DATA_ZERO_INIT static float dshotRpmAverage;
 FAST_DATA_ZERO_INIT static float dshotRpm[MAX_SUPPORTED_MOTORS];
 
 void initDshotTelemetry(const timeUs_t looptimeUs)
 {
     // if bidirectional DShot is not available
-    if (!motorConfig()->dev.useDshotTelemetry) {
+    if (!motorConfig()->dev.useDshotTelemetry && !featureIsEnabled(FEATURE_ESC_SENSOR)) {
```

Comment:
so the logic is to initialize if only esc-sensor setup as well?

Reply:
Commit 2 was suggested by @KarateBrot - have not tested yet - but agree this seems to break dshot telemetry when ESC_SENSOR is disabled.

Should be || ???

Reply:
i'm not certain of which is proper, just asking, really. attempting to think, but brain is tired.

Reply:
(!motorConfig()->dev.useDshotTelemetry && !featureIsEnabled(FEATURE_ESC_SENSOR))

vs

( ! ( motorConfig()->dev.useDshotTelemetry || !featureIsEnabled(FEATURE_ESC_SENSOR) ) )

[captain] morgan's law

Reply:
Anyway looking again we should also isolate `pt1FilterInit` execution as commited now.

Reply:
Doesn't matter if pt1FilterInit gets executed. We don't use it anyways. And you forgot the change in init.c. Take a look at my example commit 😃

Reply:
Fixed. Thanks. Overseen. Developer blindness :smiley: